### PR TITLE
Bug 1985997: wires startup monitor related controllers

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

File: pkg/operator/startupmonitorreadiness/readiness_checks.go [hunk 1]
```diff
@@ -128,11 +142,15 @@ func newPodRunning(podClient corev1client.PodInterface, monitorRevision int) fun
...
 		}
-		if len(apiServerPods.Items) != 1 {
-			return false, "PodListError", fmt.Sprintf("unable to check the pod's status, unexpected number of Kube API server pods %d, expected only one pod", len(apiServerPods.Items))
+
+		kasPodFound, kasPod, kasPodErr := findSingleMatchingPod(apiServerPods.Items, currentNodeName)
+		if kasPodErr != nil {
+			return false, "PodListError", fmt.Sprintf("unable to check the pod's status: %v", kasPodErr)
+		}
+		if !kasPodFound {
+			return false, "PodNotFound", fmt.Sprintf("unable to check the pod's status, haven't found a pod that would match the current node name %s, checked %d Kube API server pods", currentNodeName, len(apiServerPods.Items))
```
Comment (sttts) [inline pkg/operator/startupmonitorreadiness/readiness_checks.go:151]: this sounds like a dangerous situation while it can be completely normal, no?

Comment (sttts) [reply pkg/operator/startupmonitorreadiness/readiness_checks.go:150]: would something less spectacular: `static kube-apiserver pod for node %s not found yet`

File: pkg/operator/startupmonitorreadiness/readiness_checks.go [hunk 2]
```diff
@@ -128,11 +142,22 @@ func newPodRunning(podClient corev1client.PodInterface, monitorRevision int) fun
...
+		case 0:
+			return false, "PodNotFound", fmt.Sprintf("unable to check the pod's status, haven't found a pod that would match the current node name %s, checked %d Kube API server pods", currentNodeName, len(apiServerPods.Items))
+		case 1:
+			kasPod = filteredKasPods[0]
+		default:
+			podsOnCurrentNode := []string{}
+			for _, filteredKasPod := range filteredKasPods {
+				podsOnCurrentNode = append(podsOnCurrentNode, filteredKasPod.Name)
+			}
+			return false, "PodListError", fmt.Sprintf("unable to check the pod's status: found multiple pods (%v) matching the provided node name %s", podsOnCurrentNode, currentNodeName)
```
Comment (sttts) [inline pkg/operator/startupmonitorreadiness/readiness_checks.go:158]: `found multiple static kube-apiserver pods for node %s: <list of pod names>`

**Comment by sttts** [comment]
/lgtm
/approve

Follow-up:
- [ ] @p0lyn0mial EtcdHealthy -> ConnectionRefused
- [ ] @sttts early errors in NodeStatus even when not failed (CrashLooping)
- [ ] @sttts Backport of pruner to 4.8.
- [ ] @p0lyn0mial fix messages, see comments above

**Comment by p0lyn0mial** [comment]
@sttts don't forget to backport https://github.com/openshift/library-go/pull/1170

**Pull request merged**